### PR TITLE
feat(lambda-extension): Logs API processor

### PR DESCRIPTION
**Pull request by nmoutschen**
Issue #, if available:

Closes https://github.com/awslabs/aws-lambda-rust-runtime/issues/392
Closes https://github.com/awslabs/aws-lambda-rust-runtime/issues/396

Description of changes:

Register the extension with the logs API, boot an http server in a new tokio task, and process incoming logs.

By submitting this pull request

* [x] I confirm that my contribution is made under the terms of the Apache 2.0 license.
* [x] I confirm that I've made a best effort attempt to update all relevant documentation.

### CODE REVIEW

File: lambda-extension/src/logs.rs [hunk 1]
```diff
@@ -8,13 +8,104 @@ use tracing::{error, trace};
 /// See: https://docs.aws.amazon.com/lambda/latest/dg/runtimes-logs-api.html#runtimes-logs-api-msg
 #[derive(Clone, Debug, Deserialize)]
 pub struct LambdaLog {
+    /// TODO: Convert time
     /// Time when the log was generated
     pub time: String,
```
Comment (calavera) [inline lambda-extension/src/logs.rs:13]: If you add Chrono with Serde support to the dependencies, we should be able to convert the time to `DateTime<Utc>` automatically.

`chrono = { version = "^0.4", features = ["serde"] }`

**Review by calavera — CHANGES_REQUESTED** [review]
@nmoutschen can you update the README.md in the lambda-extension directory with information about the logs api support?

**Comment by nmoutschen** [comment]
> @nmoutschen can you update the README.md in the lambda-extension directory with information about the logs api support?

Done! I've added a simple example with a logs processor.

File: lambda-extension/examples/logs.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+use lambda_extension::{service_fn, Error, Extension, LambdaLog, LambdaLogRecord, SharedService};
```
Comment (calavera) [inline lambda-extension/examples/logs.rs:1]: this example is so good! 🙀

File: lambda-extension/examples/combined.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+use lambda_extension::{
```
Comment (calavera) [inline lambda-extension/examples/combined.rs:1]: another great example! 😻

**Review by calavera — APPROVED** [review]
🤘

**Pull request merged**